tests/quantization/quanto/test_quanto.py

```diff
@@ -0,0 +1,177 @@
+import unittest
+
+import torch
+
+from diffusers import (
+    QuantoConfig,
+)
+from diffusers.models.transformers.transformer_flux import FluxTransformer2DModel
+from diffusers.utils import is_optimum_quanto_available
+from diffusers.utils.testing_utils import (
+    nightly,
+    numpy_cosine_similarity_distance,
+    require_accelerate,
+    require_big_gpu_with_torch_cuda,
+    torch_device,
+)
+
+
+if is_optimum_quanto_available():
+    from optimum.quanto import QLinear
+
+
+@nightly
+@require_big_gpu_with_torch_cuda
+@require_accelerate
+class QuantoBaseTesterMixin:
+    model_id = None
+    model_cls = None
+    torch_dtype = torch.bfloat16
+    expected_memory_use_in_gb = 5
+    keep_in_fp32_module = ""
+
+    def get_dummy_init_kwargs(self):
+        return {"weights": "float8"}
```

Comment:
We could also test for `activations` and `modules_to_not_convert`. Not sure if those are being tested already.